pandas/core/dtypes/concat.py

```diff
@@ -132,6 +204,75 @@ def is_nonempty(x) -> bool:
     return np.concatenate(to_concat, axis=axis)
 
 
+def concat_arrays(to_concat):
+    """
+    Alternative for concat_compat but specialized for use in the ArrayManager.
+
+    Differences: only deals with 1D arrays (no axis keyword) and does not skip
+    empty arrays to determine the dtype.
+    In addition ensures that all NullArrayProxies get replaced with actual
+    arrays.
```

Comment:
This `concat_arrays` is basically a copied+adapated version of `concat_compat`. I initially adapted `concat_compat` to cover both cases. 

But that resulted in quite some special cases inside `concat_compat` (checking the proxy objects, while this is not needed for when `concat_compat` is used outside ArrayManager, + not skipping empties for which I added a keyword). 
So in the end I decided to split the changes in a separate function, so both `concat_compat` and `concat_arrays` are more readable / easier to follow, but it certainly gives some duplication.